### PR TITLE
dd-cdk CI/CD workflows build & CDK peer dependencies

### PR DESCRIPTION
**Pull request by zARODz11z**
### What does this PR do?
This PR adds a build Github workflow that should be ran every time a push is made to any branch on the dd-cdk repository. This PR also fixes a type error in the unit tests by casting the handler name to a string.

### Motivation
I had previously made a PR where I thought I addressed the peer dependency issue, after I merged my branch to main and attempted to make a new release I found out some of the unit tests failed, having this GitHub workflow will hopefully mitigate a faulty merge like mine in the future.

### Additional Notes
This build workflow was ported over from https://github.com/DataDog/serverless-plugin-datadog/blob/master/.github/workflows/build.yml

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "datadog-cdk-constructs",
-  "version": "0.1.6",
+  "version": "0.1.7",
```
Comment (nhinsch) [inline package.json:3]: You don't need to increment the version number here unless you are actually publishing the changes to npm as a new release. Actually, I think we would ideally change the version number in the `package.json` as part of the release script.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Node 12.7
```
Comment (nhinsch) [inline .github/workflows/build.yml:13]: For the Lint job, let's run it in the current LTS (long term support) version of Node, which is `14`.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Lint
+        run: yarn lint
+  
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        node-version: [10.17, 12.7]
```
Comment (nhinsch) [inline .github/workflows/build.yml:40]: Let's test Node version 14 here as well... we can do `10`, `12`, and `14`. If we leave out the minor version number, I believe we will just use the latest minor version within that major version, which is probably what we want here.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Build
+        run: yarn build
+
+      - name: Run tests
+        run: yarn test
+
+      - name: Upload code coverage report
```
Comment (nhinsch) [inline .github/workflows/build.yml:68]: Has this been set up? I don't know a lot about how the `codecov` tool works. But it's worth checking to see if this is working correctly, or maybe just remove this for now.

File: package.json [hunk 5]
```diff
@@ -31,9 +30,15 @@
     "typescript": "^3.9.7"
   },
   "dependencies": {
```
Comment (nhinsch) [inline package.json:32]: I thought we did not want these as dependencies any more? Just peer dependencies?

Comment (zARODz11z) [reply package.json:32]: datadog.ts imports from aws-cdk/core and the other peer dependencies. When I would remove the regular dependecies from the package.json I would get a ton of import errors from the internal files that used any of those cdk packages. It seems like just having it as a dev dependency does not add it to the local node modules therefore the dd-cdk files cannot properly import.

**Review by nhinsch — APPROVED** [review]

**Pull request merged**